src/status_im/contexts/chat/messenger/composer/events.cljs

```diff
@@ -21,6 +21,12 @@
      (when (empty? new-input)
        (mentions/clear-mentions)))))
 
+(rf/defn set-chat-input-ref
+  {:events [:chat/set-input-ref]}
+  [{:keys [db]} input-ref]
+  (let [current-chat-id (:current-chat-id db)]
+    {:db (assoc-in db [:chat/inputs current-chat-id :input-ref] input-ref)}))
```

Comment:
I know this code existed before this PR, so I'm not suggestion any change. But I would like to mention to raise awareness.

We shouldn't store component instances in the app-db. The app-db should be used to store serializable data. Also, it's a solution prone to many bugs because the ref is associated with a completely different lifecycle than the re-frame app-db lifecycle.

This may work, but it's fragile and I'm sure there's a better way.

Reply:
thank you, I'm totally agree with your statement, but this is the exception we have to do unfortunately, when it comes to some optimizations, or performance issues, the code is becoming not really nice, and we have to live with that, probably we could move ref to some global atom, but it looks even more dangerous , anyway we could discuss this separately

Reply:
Agreed :)

Reply:
There are two aspects to consider:

1. AFAIK, when `current-chat-id` changes, the screen is redrawn, so the stored input refs for the previous chat-id will get stale, so, why do we store them if they become useless? I may be wrong, could you please elaborate on it?

2. We should use `rf/reg-event-fx`

Reply:
we need it in mentions to set value `:effects/set-input-text-value [(get-in db [:chat/inputs (:current-chat-id db) :input-ref]) new-text]`